plugins/search-backend-node/src/IndexBuilder.ts

```diff
@@ -76,18 +76,18 @@ export class IndexBuilder {
    * the decorator, it will be applied to documents from all known collators,
    * otherwise it will only be applied to documents of the given types.
    */
-  addDecorator({ decorator }: RegisterDecoratorParameters): void {
-    const types = decorator.types || ['*'];
+  addDecorator({ factory }: RegisterDecoratorParameters): void {
```

Comment:
similar comment as above

Just one thing that came to mind when reading this, did you ever thing of changing this to something like addDecoratorFactory, as its more a decorator factory that gets aded now than a collator? 🤔

Reply:
(Closing this thread out and consolidating discussion on the collator comment)